crates/core/src/protocols/sumcheck/prove/eq_ind.rs

```diff
@@ -247,6 +247,14 @@ where
 		Ok(self)
 	}
 
+	/// Automatically infers a constant suffix consisting of zeroes if not explicitly specified.
+	pub fn with_auto_zero_suffixes(mut self) -> Self {
```

Comment:
This seems sort of suboptimal compared with a strategy of passing this information from the frontend, where it's already available. The witness generation code should set this field explicitly on a witness column, instead of requiring a traversal of all the evaluations.

Reply:
We're punting on this for now.